e2e/test/scenarios/visualizations-tabular/smartscalar-trend.cy.spec.js

```diff
@@ -1,15 +1,190 @@
+      },
+      { visitQuestion: true },
+    );
+
+    cy.findByTestId("viz-settings-button").click();
+    cy.findByTestId("chartsettings-sidebar").findByText("Data").click();
+
+    // primary number
+    cy.findByTestId("scalar-container").findByText("344");
+    cy.findByTestId("chartsettings-sidebar").findByText("Count").click();
+    popover().within(() => {
+      // should only have two options
+      cy.findAllByRole("option").should("have.length", 2);
+
+      // selected should be highlighted
+      cy.findByLabelText("Count").should("have.attr", "aria-selected", "true");
+
+      // should not be highlighted b/c not selected
+      cy.findByLabelText("Sum of Total").should(
+        "have.attr",
+        "aria-selected",
+        "false",
+      );
+      cy.findByText("Sum of Total").click();
+    });
+    cy.findByTestId("scalar-container").findByText("30,759.47");
+
+    // comparisons
+    // default should be previous period (since we have a dateUnit)
+    cy.findByTestId("scalar-container").findByText("30,759.47");
+    cy.findByTestId("scalar-previous-value").within(() => {
+      cy.findByText("vs. previous month:");
+      cy.findByText("45,683.68");
+    });
+
+    // previous value
+    cy.findByTestId("chartsettings-sidebar")
+      .findByText("Previous month")
+      .click();
+    menu().findByText("Previous value").click();
+    cy.findByTestId("scalar-previous-value").within(() => {
+      cy.findByText("vs. Mar:");
+      cy.findByText("45,683.68");
+    });
+
+    // periods ago
+    cy.findByTestId("chartsettings-sidebar")
+      .findByText("Previous value")
+      .click();
+    menu().within(() => {
+      // should clamp over input to maxPeriodsAgo
+      cy.get("input").click().type("100{enter}");
+      cy.get("input").should("have.value", 48);
+
+      // should clamp under input to 2
+      cy.get("input").click().type("-45{enter}");
+      cy.get("input").should("have.value", 2);
+
+      // should not allow invalid input
+      cy.get("input").click().type("3.90293{enter}");
+      cy.get("input").should("have.value", 2);
+
+      // should allow valid input
+      cy.get("input").click().type("3{enter}");
+    });
+    cy.findByTestId("scalar-previous-value").within(() => {
+      cy.findByText("vs. Jan:");
+      cy.findByText("52,249.59");
+    });
+  });
+
+  it("should allow display settings to be changed and display should reflect changes", () => {
+    cy.createQuestion(
+      {
+        name: "13710",
+        query: {
+          "source-table": ORDERS_ID,
+          aggregation: [["count"], ["sum", ["field", ORDERS.TOTAL, null]]],
+          breakout: [
+            ["field", ORDERS.CREATED_AT, { "temporal-unit": "month" }],
+          ],
+        },
+        display: "smartscalar",
+      },
+      { visitQuestion: true },
+    );
+
+    // scalar.switch_positive_negative setting
+    cy.icon("arrow_down").should(
+      "have.css",
+      "color",
+      Color(colors.error).string(),
+    );
```

Comment:
My gut tells me we shouldn't be asserting the colors, but I can't come up with a better alternative
@alxnddr WDYT?

Reply:
I think it is fine to check styles in such cases because it is a reflection of the code logic. If possible, would be nice to test in isolation — in jest specs